After importing `polaris-react` into my project, my App's `tsc` type check fails.

<!-- For feature requests, please use the following template: https://github.com/Shopify/polaris/issues/new?labels=Feature+request&template=FEATURE_REQUEST.md -->

## Issue summary

<!--
Write a short description of the issue here ↓
-->
Hi there, I'm on `polaris@11.19.0`, but when I run a `tsc` (Typescript compiler) on my app, I'm getting the following error:
```
node_modules/@shopify/polaris/build/ts/src/utilities/use-theme.d.ts:408:44 - error TS2307: Cannot find module '@shopify/polaris-tokens/dist/types/src/themes/types' or its corresponding type declarations.

408 export declare function useTheme(): import("@shopify/polaris-tokens/dist/types/src/themes/types").ExtractMetaThemeValues<{
```
I do have `compilerOptions.skipLibCheck` set to `true` in my App's `tsconfig.json`, and if I turn that off, I can workaround this issue, but it's not ideal.  The issue seems to be due to `ExtractMetaThemeValues` and `MetaTokenProperties` not being exported from `polaris-tokens`, and `polaris-react` is trying to reach directly into the non-exported type, which is fine for `rollup` (which seems to be `polaris`'s builder), but not for `tsc`.

Another workaround I found was to set the following values in my `tsconfig.json`, which fixes the pathing issue:
```
{
  "compilerOptions": {
    "skipLibCheck": false,
    "paths": {
      "@shopify/polaris-tokens/dist/types/src/themes/types": ["node_modules/@shopify/polaris-tokens/dist/types/src/themes/types.ts"],
    }
  }
}
```

A true solution I've found is to set the [following lines](https://github.com/Shopify/polaris/blob/main/polaris-tokens/src/index.ts#L11) in `polaris-tokens` to:
```
export type {
  ThemeName,
  Theme,
  ExtractMetaThemeValues,
  MetaTokenProperties,
} from './themes/types';
```
which changes the built path to a correct value.  I'll open a PR with the above fix, where we can discuss if this is the approach you want to go with.

## Expected behavior

<!--
What do you think should happen?
-->
I should be able to import `polaris` into my app, and compile my Typescript types with `compilerOptions.skipLibCheck = false`.

## Actual behavior

<!--
What actually happens?

Tip: include an error message (in a `<details></details>` tag) if your issue is related to an error while running Polaris.

If you include an animated gif showing your issue, wrapping it in a details tag is also recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>

-->
When running `tsc`, I get the following:
```
node_modules/@shopify/polaris/build/ts/src/utilities/use-theme.d.ts:408:44 - error TS2307: Cannot find module '@shopify/polaris-tokens/dist/types/src/themes/types' or its corresponding type declarations.

408 export declare function useTheme(): import("@shopify/polaris-tokens/dist/types/src/themes/types").ExtractMetaThemeValues<{
```

## Steps to reproduce the problem

1. Install `polaris` into Typescript app.
2. Make sure `compilerOptions.skipLibCheck` in the app's `tsconfig.json` is set to `false`.
3. Run `tsc`.

## Reduced test case

The best way to get your bug fixed is to provide a reduced test case. This [CodeSandbox template](https://codesandbox.io/s/q82mlq0m26) is a great starting point.

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: 11.19.0

Or run `npx envinfo --system --binaries --browsers --npmPackages react,react-dom,@shopify/polaris` to provide specifications on your environment including version numbers, browser, device, and operating system.

Paste the results here:

```bash
  System:
    OS: Linux 6.2 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (2) x64 Intel(R) Xeon(R) CPU @ 2.60GHz
    Memory: 11.70 GB / 15.61 GB
    Container: Yes
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.20.2 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/src/github.com/Shopify/polaris/node_modules/.bin/yarn
    npm: 8.19.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/bin/watchman
```

--- a/.changeset/poor-poems-thank.md
+++ b/.changeset/poor-poems-thank.md
@@ -1,0 +1,5 @@
+---
+'@shopify/polaris-tokens': patch
+---
+
+Export types missing for the Typescript Compiler (tsc) when using Node16/Bundler Module Resolution.

--- a/polaris-tokens/src/index.ts
+++ b/polaris-tokens/src/index.ts
@@ -15,6 +15,7 @@ export {
 export {metaThemes, metaThemeDefault} from './themes';
 
 export type {
+  ExtractMetaThemeValues,
   MetaTheme,
   MetaThemeShape,
   MetaTokenGroupShape,

